### DISCUSSION TITLE
Configuration rule_files is not available.

### DISCUSSION THREAD

**Discussion by yixiubest** [prometheus/prometheus#11866] [Q&A]
### What did you do?

Ignore my poor English, plz.
1.I volume a custom rule-config path (/etc/config/custom-rules),like this:
![image](https://user-images.githubusercontent.com/38375081/212799077-19bd360e-5239-4331-8c95-57a271c6c982.png)

The pvc is provided by nfs.

2.And I configured the rule_files like this:
![image](https://user-images.githubusercontent.com/38375081/212800176-5f90b7a0-005a-4bbe-a752-0686f6be40db.png)

3.Then I created the rule file on the nfs and found it in the path custom-rules, and I have executed the reload url, but it showed an error:
![image](https://user-images.githubusercontent.com/38375081/212799941-2ebfcde3-af00-40c4-aa54-bfd85e04728d.png)

The rule file like this:
![image](https://user-images.githubusercontent.com/38375081/212800357-27877cd7-48b5-4d7b-9a20-2b55be88e87b.png)

### What did you expect to see?

Why it showed the error? Is there any inappropriate configuration? 
And if I want to custom rules what shuld I do, I have tried to find anwser by the Docs, but failed. SOS plz.

### What did you see instead? Under which circumstances?

Error like up-show

### System information

k8s，prometheus-v2.26.0

### Prometheus configuration file

```yaml
see up-show
```

**Comment by roidelapluie** [comment]
It would be helpful if you could check the custom rule file using Prometheus's built-in rule parser tool "promtool". This can help to identify any syntax errors in the rule file. The command for checking the rule file with promtool is:

```
promtool check rules /etc/config/custom-rules/test_0001.yml
```

This will provide detailed output of any errors or warnings found in the custom rule file, making it easier to identify and correct any issues.

Also, it's important to check the main Prometheus configuration file and ensure that the path to the custom rule file is correct, and that Prometheus has permission to access it.

```
promtool check config /etc/config/prometheus.yml
```

**Reply by yixiubest** [reply]
thank you very much

**Comment by yixiubest** [comment]
![image](https://user-images.githubusercontent.com/38375081/218961778-a40ce022-58fb-46fb-bd33-e8388396fc70.png)

@roidelapluie After I corrected my previous mistakes，The pod prometheus-server can run start normally，but a fews minutes later，the system tries to create a new pod.

![image](https://user-images.githubusercontent.com/38375081/218962965-d511bca0-8788-4a44-80f1-3c06f7edac8c.png)

But because of the lock file, the new pod created failed


![image](https://user-images.githubusercontent.com/38375081/218960515-f16db8ea-640a-455e-aae6-b6c873761014.png)

rule file:
![image](https://user-images.githubusercontent.com/38375081/218963639-2dff4316-9c76-48fa-85fd-4da4b3de1a9a.png)

When I check the rule config, It reports an mistake like the pic showing，that looks like the format of yml is incorrect，but I can't found
anything wrong with my rule file like the pic showing.

So my questions are:
1.In addition to deleting the lock file, is there any way to make the creation of the new pod successful？
2.Is there anything wrong with the rule file? Why it shown me unmarshal errors?

I can't find the problem, so I need your help please.
